sysutils/john/files/patch-apple-m1.diff

```diff
@@ -0,0 +1,195 @@
+--- src/configure
++++ src/configure
+@@ -9321,6 +9321,42 @@ else
+      { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
+ $as_echo "no" >&6; }
+ 
++fi
++rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
++  conftest.$ac_objext conftest.beam conftest.$ac_ext
++fi
++
++  CC="$CC_BACKUP -march=armv8-a+simd"
++  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for ASIMD" >&5
++$as_echo_n "checking for ASIMD... " >&6; }
++  if test "$cross_compiling" = yes; then :
++  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
++$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
++as_fn_error $? "cannot run test program while cross compiling
++See \`config.log' for more details" "$LINENO" 5; }
++else
++  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
++/* end confdefs.h.  */
++
++    #include <arm_neon.h>
++  #include <stdio.h>
++  extern void exit(int);
++  int main(){uint32x4_t t;*((long*)&t)=1;t=veorq_u32(t,t);if((*(unsigned*)&t)==88)printf(".");exit(0);}
++
++
++_ACEOF
++if ac_fn_c_try_run "$LINENO"; then :
++  CPU_BEST_FLAGS="-march=armv8-a+simd"
++     SIMD_NAME="ASIMD"
++     { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
++$as_echo "yes" >&6; }
++     ARCH_LINK=arm64le.h
++
++else
++  CPU_NOTFOUND="1"
++     { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
++$as_echo "no" >&6; }
++
+ fi
+ rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
+   conftest.$ac_objext conftest.beam conftest.$ac_ext
+@@ -9329,6 +9365,7 @@ fi
+ else
+   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for NEON" >&5
+ $as_echo_n "checking for NEON... " >&6; }
++  CC="$CC_BACKUP -mfpu=neon"
+   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+ /* end confdefs.h.  */
+ 
+@@ -9350,6 +9387,34 @@ else
+      { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
+ $as_echo "no" >&6; }
+ 
++fi
++rm -f core conftest.err conftest.$ac_objext \
++    conftest$ac_exeext conftest.$ac_ext
++  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for ASIMD" >&5
++$as_echo_n "checking for ASIMD... " >&6; }
++  CC="$CC_BACKUP -march=armv8-a+simd"
++  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
++/* end confdefs.h.  */
++
++    #include <arm_neon.h>
++  #include <stdio.h>
++  extern void exit(int);
++  int main(){uint32x4_t t;*((long*)&t)=1;t=veorq_u32(t,t);if((*(unsigned*)&t)==88)printf(".");exit(0);}
++
++
++_ACEOF
++if ac_fn_c_try_link "$LINENO"; then :
++  CPU_BEST_FLAGS="-march=armv8-a+simd"
++     SIMD_NAME="ASIMD"
++     ARCH_LINK=arm64le.h
++     { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
++$as_echo "yes" >&6; }
++
++else
++  CPU_NOTFOUND="1"
++     { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
++$as_echo "no" >&6; }
++
+ fi
+ rm -f core conftest.err conftest.$ac_objext \
+     conftest$ac_exeext conftest.$ac_ext
+diff --git src/m4/jtr_arm_logic.m4 src/m4/jtr_arm_logic.m4
+index 3703e74b02..de47735cc2 100644
+--- src/m4/jtr_arm_logic.m4
++++ src/m4/jtr_arm_logic.m4
```

Comment:
> Unfortentnly I can't use `use_autoreconf` here. 

I wasn't suggesting that you should; in fact, I said the opposite. I was just pointing out that if you happened to use autoreconf, then it would work correctly, the only problem would be wasted space by providing an unneeded patch for the configure script (since the patched configure script would immediately be discarded and regenerated).

> Anyway, I just grab the commit as is and it applied and works without any issues => seems good enough to keep it what way.

Yes, I'm sure that's what you did, and I'm sure it worked on your system. What I was pointing out is that by providing patches for both the configure script and the files from which it is generated, in that order, a type of [race condition](https://en.wikipedia.org/wiki/Race_condition) can occur which, depending on timing, can make the build happen differently: in that case, the build would automatically run autoreconf. That may or may not fail, depending on whether the user happens to have a working autoconf installed, but it is not [reproducible](https://trac.macports.org/wiki/ReproducibleBuilds) and timing-related bugs can be difficult to identify, so I was trying to save you future debugging time by pointing this one out in advance.

> And to be on safe side I've removed all `.m4` parts. Only `./configure` and code changes.

Thanks. That should ensure the problem doesn't occur.